### PR TITLE
Small grammar fixes and wording suggestions

### PR DESCRIPTION
**Pull request by Dhghomon**
Just a small PR with some grammar fixes and wording suggestions I've been jotting down bit by bit over the past few weeks.

### CODE REVIEW

File: docs/edgeql/paths.rst [hunk 1]
```diff
@@ -37,7 +37,7 @@ friend of another User*.
 
   select User.friends;
 
-Paths can traverse arbitrarily many links.
+Paths can arbitrarily traverse multiple links.
```
Comment (raddevon) [inline docs/edgeql/paths.rst:40]: I agree this needs a rewrite, but I think this one may change the intended meaning slightly. What about this?

```suggestion
Paths can traverse to an arbitrary depth in a series of nested links.
```

Judging based on the example, I think this gets closer to what we're trying to show.

File: docs/intro/schema.rst [hunk 2]
```diff
@@ -226,7 +226,8 @@ understand backlink syntax is to split it into two parts:
 
 ``[is Movie]``
   This is a *type filter* that filters out all objects that aren't ``Movie``
-  objects.
+  objects. A backlink still works without this filter, but could contain any 
+  other number of objects besides `` Movie``s.
```
Comment (raddevon) [inline docs/intro/schema.rst:230]:
```suggestion
  other number of objects besides `` Movie`` objects.
```

Sphinx doesn't seem to cooperate when you have inline code butted up against another character. I usually just write around it like this.

**Review by raddevon — CHANGES_REQUESTED** [review]
Two minor suggestions. The rest looks good!

File: docs/intro/schema.rst (hunk 2 above)
Comment (Dhghomon) [reply docs/intro/schema.rst:230]: Ah! Good to know.

**Review by raddevon — APPROVED** [review]

**Pull request merged**